src/Filters/molecules/FiltersResults.tsx

```diff
@@ -61,13 +64,50 @@ const FilterLabelsGroup = styled.div`
   margin-left: 10px;
 `;
 
+const validateDateFormat = (resultsDateFormat: string): boolean => {
+  let isFormatValid = true;
+
+  if (resultsDateFormat) {
+    try {
+      const tryDate = new Date();
+      format(tryDate, resultsDateFormat);
+    } catch (error) {
+      isFormatValid = false;
+    }
+  }
+
+  return isFormatValid;
+};
+
 const renderResults = (template: string, total: number) => {
   return template.replace('[TOTAL_RESULTS]', `${total}`);
 };
 
+const renderLabel = (item: IFilterItem | DateInterval | Date, icon?: string, filterName?: string, resultsDateFormat?: string) => {
+
+  let textLabel: string = "";
+  const isDateFormatValid = resultsDateFormat ? validateDateFormat(resultsDateFormat) : false;
+
+  if (filterName && isFilterItem(item)) {
+    textLabel = `${filterName}: ${item.text}`;
+  } else if (filterName && item instanceof Date) {
+    textLabel = isDateFormatValid && resultsDateFormat ? `${filterName}: ${format(item, resultsDateFormat)}` : `${filterName}: ${item.toDateString()}`;
+  } else if (filterName && isDateInterval(item)) {
+    textLabel = isDateFormatValid && resultsDateFormat ? `${filterName}: ${format(item.start, resultsDateFormat)} - ${format(item.end, resultsDateFormat)}` : `${filterName}: ${item.start.toDateString()} - ${item.end.toDateString()}`;
+  } else if (!filterName && isFilterItem(item)) {
+    textLabel = item.text;
+  } else if (!filterName && item instanceof Date) {
+    textLabel = isDateFormatValid && resultsDateFormat ? format(item, resultsDateFormat) : item.toDateString();
+  } else if (!filterName && isDateInterval(item)) {
+    textLabel = isDateFormatValid && resultsDateFormat ? `${format(item.start, resultsDateFormat)} - ${format(item.end, resultsDateFormat)}` : `${item.start.toDateString()} - ${item.end.toDateString()}`;
```

Comment:
Can we break these across a few lines. It’s longer so hard to read and the : within the values make it hard to see the one that is used in the ternary. I thought it was a nested ternary at first.

Reply:
I have updated the logic to be easier to understand.